### PR TITLE
Add Conv1D implementation

### PR DESCRIPTION
**Pull request by nivibilla**
Hey,

I added this code for project I was doing. Thought might aswell merge it if it doesn't break anything. 

Thanks!

### CODE REVIEW

**Comment by nivibilla** [comment]
@microsoft-github-policy-service agree

**Comment by edwardjhu** [comment]
Thanks for contributing, Srinivas!

The code for Conv1d seems identical to Conv2d. It might make sense to abstract it further for all Conv layers. Would you like to take a stab at that? I'm happy to work with you to build some test cases and get it merged.

**Comment by nivibilla** [comment]
Sure. I can have a crack at it

**Comment by nivibilla** [comment]
Hey, I've added it to the 'abstraction' branch of my fork

**Comment by heitorrapela** [comment]
Hello @nivibilla, you should not update pip install in the README file from the repo to you fork.

**Comment by nivibilla** [comment]
@heitorrapela thanks for pointing that out. I forgot to change. Ive also implemented the abstraction now

**Comment by edwardjhu** [comment]
merged. thanks for contributing!

**Comment by GT-KIM** [comment]
Hello,
I think this branch is not for Conv1d.
lora_A @ lora_B has shape: (out_channel * kernel_size, in_channel_kernel_size).
However, conv.weight has shape: (out_channel, in_channel, kernel_size).
I think this is only for Conv2d.

**Comment by RasmusD** [comment]
Hi,
I actually think this change breaks a lot of things. The primary change is that the resulting layer will have the convolution weight and bias under module.conv.weight and module.conv.bias - this is incompatible with default convolutional layers which have weights and bias as module.weight and module.bias.
This requires different loading logic for the convolution layers and breaks other things like potential normalisation methods like spectral normalisation that expects those module in certain places.

**Comment by lastapple** [comment]
> Hi, I actually think this change breaks a lot of things. The primary change is that the resulting layer will have the convolution weight and bias under module.conv.weight and module.conv.bias - this is incompatible with default convolutional layers which have weights and bias as module.weight and module.bias. This requires different loading logic for the convolution layers and breaks other things like potential normalisation methods like spectral normalisation that expects those module in certain places.

I quite agree. Since I used "torch.nn.utils.weight_norm" in my network, I had to make many changes to the source code to run my projects.
